application/helpers/expressions/em_core_helper.php

```diff
@@ -3250,3 +3251,74 @@ function exprmgr_unique($args)
     }
     return true;
 }
+
+/**
+ * Takes a JSON encoded string and extracts a value as a string using any additional arguments as indexes to navigate the JSON object.
+ * Examples:
+ *	If
+ *		$json_encoded_string='{"something":{"new":"a surprise!","old":"boring","complex":{"older":"boringer"},"array":["item 1","item 2"],"numeric":56}}'
+ *	Then
+ * 		exprmgr_json_value($json_encoded_string,"something","new")="a surprise!"
+ * 		exprmgr_json_value($json_encoded_string,"something","old")="boring"
+ * 		exprmgr_json_value($json_encoded_string,"something","complex")='{"older":"boringer"}'
+ * 		exprmgr_json_value($json_encoded_string,"something","array")='["item 1","item 2"]'
+ * 		exprmgr_json_value($json_encoded_string,"something","array[1]")="item 2"
+ * 		exprmgr_json_value($json_encoded_string,"something","array[50][1]")="item 2"
+ * 		exprmgr_json_value($json_encoded_string,"something","missing")=""
+ * 		exprmgr_json_value($json_encoded_string,"something","numeric")="56"
+ * 		exprmgr_json_value($json_encoded_string,"something")='{"new":"a surprise!","old":"boring","complex":{"older":"boringer"},"array":["item 1","item 2"],"numeric":56}'
+ * 		exprmgr_json_value($json_encoded_string)='{"something":{"new":"a surprise!","old":"boring","complex":{"older":"boringer"},"array":["item 1","item 2"],"numeric":56}}'
+ * @author Leon Janzen
+ * 
+ * @param string $json_encoded_string A JSON encoded string. Required
+ * @param string $index1 The first level object name. Optional
+ * @param string $index2 The second level object name. Optional
+ * @param string $index3 The third level object name. Optional
+ * etc.
+ * @return string
+ */
+function exprmgr_json_value(string $json_encoded_string, string $index1=null) {
+	
+	$json_element = null;
+	$json_array = json_decode($json_encoded_string,true);
+
+	if(is_null($json_array)) { //Check if the string was not valid JSON, and could not be decoded.
+		//I'm not sure which is preferable, to return an error message, or throw an error, or do nothing and just return null.
+		
+		//Return an error message as the value.
+		//$json_element = "Invalid JSON (json_decode() failed).";
+
+		//Throw an error.
+		throw new \ErrorException("Invalid JSON (json_decode() failed).");
```

Comment:
😵 there are a system to get errors https://github.com/LimeSurvey/LimeSurvey/blob/297841a92f4e0e52e5a9362d73665eb5a00df4b6/application/helpers/expressions/em_core_helper.php#L2229

Reply:
Thanks for taking the time to review my code. I wasn't aware that this functionality can be accomplished through a plugin, so I'll pursue that option instead.